input-profile.ts

```diff
@@ -0,0 +1,272 @@
+import {Component, Object3D} from '@wonderlandengine/api';
+import {HandTracking} from './hand-tracking.js';
+import {vec3, quat} from 'gl-matrix';
+import {property} from '@wonderlandengine/api/decorators.js';
+
+const _TempVec = vec3.create();
+const _TempQuat = quat.create();
+const _TempRotation1 = new Float32Array(4);
+const _TempRotation2 = new Float32Array(4);
+const minTemp = new Float32Array(3);
+const maxTemp = new Float32Array(3);
+const hands = ['left', 'right'];
+
+interface VisualResponse {
+    target: Object3D;
+    min: Object3D;
+    max: Object3D;
+    id: number;
+}
+
+export class InputProfile extends Component {
+    static TypeName = 'input-profile';
+    static Cache: Map<string, any> = new Map();
+    private _gamepadObjects: Record<string, Object3D> = {};
+    private _controllerModel: Object3D | null = null;
+    private _defaultControllerComponents: Component[] | undefined;
+    private _handedness!: string;
+    private _profileJSON: any = null;
+    private _modelLoaded!: boolean;
+    private _gamepad: Gamepad | undefined;
+    private _buttons: VisualResponse[] = [];
+    private _axes: VisualResponse[] = [];
+
+    url!: string;
+    toFilter: Set<string> = new Set(['vr-mode-active-mode-switch']);
+
+    @property.object()
+    defaultController!: Object3D;
+
+    @property.enum(hands, 0)
+    handednessIndex: number = 0;
+
+    @property.string(
+        'https://cdn.jsdelivr.net/npm/@webxr-input-profiles/assets@latest/dist/profiles/'
+    )
+    path!: string;
+
+    @property.string()
+    customProfileFolder!: string;
+
+    @property.bool(false)
+    mapToDefaultController!: boolean;
+
+    @property.object()
+    trackedHand!: Object3D;
+
+    start() {
+        this._controllerModel = null;
+        this.toFilter = new Set(['vr-mode-active-mode-switch']);
+        this._defaultControllerComponents = this.getComponents(this.defaultController);
+        this._handedness = hands[this.handednessIndex];
+
+        this.engine.onXRSessionStart.add(() => {
+            console.warn('adding event listener for inputChange');
+            this.engine.xr?.session.addEventListener(
+                'inputsourceschange',
+                this.onInputSourcesChange.bind(this)
+            );
+        });
+    }
+
+    onDeactivate() {
+        this.engine.onXRSessionStart.add(() => {
+            this.engine.xr?.session.removeEventListener(
+                'inputsourceschange',
+                this.onInputSourcesChange.bind(this)
+            );
+        });
+    }
+
+    setHandTrackingControllers(controllerObject: Object3D) {
+        const HandtrackingComponent = this.trackedHand.getComponent(HandTracking);
+        if (!HandtrackingComponent) return;
+        /**@ts-ignore**/
```

Comment:
Instead of using `ts-ignore`, do:

```ts
/** @todo: Remove any when hand tracking is typed. */
(component as any).controllerToDeactivate = controllerObject;
```